src/pytorch_lightning/strategies/tpu_spawn.py

```diff
@@ -114,30 +114,32 @@ def _validate_patched_dataloaders(model: "pl.LightningModule") -> None:
             if not source.is_module():
                 TPUSpawnStrategy._validate_dataloader(source.instance)
```

Comment:
I have no idea to decide it.
source.instance can be `TRAIN_DATALOADERS = Union[
    DataLoader,
    Sequence[DataLoader],
    Sequence[Sequence[DataLoader]],
    Sequence[Dict[str, DataLoader]],
    Dict[str, DataLoader],
    Dict[str, Dict[str, DataLoader]],
    Dict[str, Sequence[DataLoader]]
]`
but _validate_dataloader can't support dictionary.

Reply:
@BongYang I think in this case we want to raise an error here if it is a dict. And then we can open an issue if we would like to support it in the future. What do you think?

Reply:
I pushed a commit with a proposed change to make the validation match the type of the dataloaders well.

Reply:
thank you!